modules/standard/OS.chpl

```diff
@@ -172,9 +172,6 @@ module OS {
     inline operator :(x:suseconds_t, type t:c_int)
       return __primitive("cast", t, x);
     pragma "no doc"
-    inline operator :(x:c_int, type t:suseconds_t)
-      return __primitive("cast", t, x);
-    pragma "no doc"
```

Comment:
Hmm. POSIX doesn't seem to say what the type `suseconds_t` is at all, other than it can store 1000000. Meanwhile, there is a cast just above going the other way, `suseconds_t` to `c_int`.

I think my preference would be to solve this in a different way. Some ideas:
 * only have casts between Chapel `int` and these types and back
 * Or, use where clauses to handle the case that `c_int == c_long`
 * use only `integral` for the Chapel type to cast to/from e.g. above `inline operator :(x:off_t, type t:integral)`
 * Or, just remove the `c_long` casts for `suseconds_t` since that numeric types only need to store up to 1000000 .

The situation is a bit different for `time_t` and indeed it might match `c_int`, `c_long`, or even `c_float` or `c_double`. So in that case it would be just the 1st two options.

Anyway, it seems that this problem is present throughout this module in other functions as well. I think of all of the above options, my favorite is to simply use `:integral` for the Chapel types on these, and then remove the overloads for other numeric types. `c_int` / `c_long` should be handled by the `integral` case even if it is the only case.